kube/pod_test.go

```diff
@@ -2047,28 +2066,191 @@ func TestPodCPUAHelmActive(t *testing.T) {
 	`, actual)
 }
 
-func TestGetSecurityContext(t *testing.T) {
+func TestGetSecurityContextCapList(t *testing.T) {
 	assert := assert.New(t)
 
 	role := podTemplateTestLoadRole(assert)
 	if role == nil {
 		return
 	}
 
-	sc := getSecurityContext(role)
-	if !assert.NotNil(sc) {
+	t.Run("Kube", func(t *testing.T) {
+		sc := getSecurityContext(role, false)
+		if !assert.NotNil(sc) {
+			return
+		}
+
+		actual, err := testhelpers.RoundtripKube(sc)
+		if !assert.NoError(err) {
+			return
+		}
+		testhelpers.IsYAMLEqualString(assert, `---
+			capabilities:
+				add:
+				-	"SOMETHING"
+		`, actual)
+	})
+
+	t.Run("Helm", func(t *testing.T) {
+		sc := getSecurityContext(role, true)
+		if !assert.NotNil(sc) {
+			return
+		}
+
+		t.Run("OverrideNone", func(t *testing.T) {
+			config := map[string]interface{}{
+				"Values.sizing.myrole.capabilities": []interface{}{},
+			}
+			actual, err := testhelpers.RoundtripNode(sc, config)
+			if !assert.NoError(err) {
+				return
+			}
+			testhelpers.IsYAMLEqualString(assert, `---
+				capabilities:
+					add:
+					-	"SOMETHING"
+			`, actual)
+		})
+
+		t.Run("OverrideALL", func(t *testing.T) {
+			config := map[string]interface{}{
+				"Values.sizing.myrole.capabilities": []interface{}{"ALL"},
+			}
+			actual, err := testhelpers.RoundtripNode(sc, config)
+			if !assert.NoError(err) {
+				return
+			}
+			testhelpers.IsYAMLEqualString(assert, `---
+				privileged: true
+			`, actual)
+		})
+
+		t.Run("OverrideSomething", func(t *testing.T) {
+			config := map[string]interface{}{
+				"Values.sizing.myrole.capabilities": []interface{}{"something"},
+			}
+			actual, err := testhelpers.RoundtripNode(sc, config)
+			if !assert.NoError(err) {
+				return
+			}
+			testhelpers.IsYAMLEqualString(assert, `---
+				capabilities:
+					add:
+					-	"SOMETHING"
+					-	"something"
+			`, actual)
+		})
+	})
+}
+
+func TestGetSecurityContextNil(t *testing.T) {
+	assert := assert.New(t)
+
+	role := podTemplateTestLoadRole(assert)
+	if role == nil {
 		return
 	}
 
-	actual, err := testhelpers.RoundtripKube(sc)
-	if !assert.NoError(err) {
+	// Clear the capability list from the manifest to force fully-privileged mode.
```

Comment:
Umm, doesn't that force _minimally_ privileged mode instead?

Reply:
Right. And actually `default privileges`, might not be minimal. Will fix.

Reply:
Done with commit b5563f564ddf71072c25c524d9d4c29e43733e18